### PR TITLE
feat: improve UX with vim.notify

### PR DESCRIPTION
**Pull request by tami5**
Recently vim.notify has been getting popular and tools such as [rcarriga/nvim-notify] enable the user to customize the looks and feel of message coming from user setup. Thus, replacing error and print with a wrapper around vim.notify will greatly improve telescope user experience.

cc @Conni2461 @fdschmidt93 @tjdevries

edit 1:

After few tests, it turned out there is an issue hidden I need to report 😆

[asciinema rec](https://asciinema.org/a/jDSKoXcGJUReddyMcMpD4ZlyT)

[rcarriga/nvim-notify]: https://github.com/rcarriga/nvim-notify

### CODE REVIEW

**Comment by tami5** [comment]
PR title needs to be `feat: improve UX with vim.notify` but my setup broke it because of the branch name and auto rename feature I have going on with GitHub 😅

**Comment by Conni2461** [comment]
i already mentioned a couple of times that this is a good idea. thanks for doing it. ping me if you need a review :) i am currently in exam phase but i try to review it

**Comment by Conni2461** [comment]
@tami5 i just rebased this and would finish it. Can you tell me the big thing that was missing?

File: lua/telescope/actions/init.lua [hunk 1]
```diff
@@ -26,10 +26,23 @@ local resolver = require "telescope.config.resolve"
...
 local actions = setmetatable({}, {
   __index = function(_, k)
-    error("Key does not exist for 'telescope.actions': " .. tostring(k))
+    error() -- really that what it took to slient it?
+    -- utils.notify("actions", {
+    --   msg = ("'%s' does not exist."):format(tostring(k)),
+    --   level = "TRACE",
+    --   panic = true,
+    --   report = true,
+    -- })
```
Comment (Conni2461) [inline lua/telescope/actions/init.lua:35]: TODO @Conni2461

**Comment by tami5** [comment]
> @tami5 i just rebased this and would finish it. Can you tell me the big thing that was missing?

Thanks conni. I believe stack tracing is what's missing and maybe steel some debug studf from plenary.log

**Comment by Conni2461** [comment]
I am not sure about the `panic` because it first panics and after you dissmiss that you get the notification. Maybe for that we should just use `log.error(debug.traceback ...)`

**Comment by tami5** [comment]
LGTM.

**Comment by Conni2461** [comment]
Thanks for the PR tami :)

**Pull request merged**